### PR TITLE
Related items and comparison

### PR DESCRIPTION
**Pull request by mrdooby**
Added thumbnails (first img, not default) to card components, added star ratings to card, comparison modal names rendered dynamically

### CODE REVIEW

File: client/src/Components/RelatedItemsAndComparison/RelatedCards.jsx [hunk 1]
```diff
@@ -3,33 +3,61 @@ import axios from 'axios';
...
+        if (props.reviewData[i].product_id == id) {
+          return props.reviewData[i].ratings
+        }
+      }
+    }
+  }
   return (
     <>
     {products?.map((prod) => {
+      {console.log('url', getUrl(prod.id))}
```
Comment (Panlord) [inline client/src/Components/RelatedItemsAndComparison/RelatedCards.jsx:45]: The url gets console logged many times upon updating the selected product. Would you mind removing the console log? That is, only if it is unnecessary. I don't know if you want to keep it for future use.

File: client/src/Components/RelatedItemsAndComparison/RelatedCards.jsx [hunk 2]
```diff
@@ -3,33 +3,61 @@ import axios from 'axios';
...
+              {modal ? <ComparisonModal modalToggle = {modalToggle} clickedProd={clickedProd} chosenProduct={props.chosenProduct}/> : null }
+              <div onClick={(e) => {handleCardClick(prod.id)}}>
+              <ThumbnailImage src={getUrl(prod.id)} />
+              <p>{prod.category}</p>
+              <p>{prod.name}</p>
+              <p>{prod.default_price}</p>
+              <StarContainer>
+                <StarRating reviewData={getRatings(prod.id)}/>
+              </StarContainer>
+            </div>
```
Comment (Panlord) [inline client/src/Components/RelatedItemsAndComparison/RelatedCards.jsx:60]: Your modal here is implemented nicely. Note that when the modal pops up, we can still see the filled stars in all of the shared star rating components. I presume it's due to the z-index setting of said stars. Nothing serious here, just a small styling thing you may want to look at (feel free to play around with the z-index of the said star rating shared component if need be).

**Review by Panlord — APPROVED** [review]
I've pulled your changes to my local machine and they work fine when tested (unstressed). I've left two comments-- please take a look, time permitting. 

Nonetheless, solid work. Approved for merge.

**Comment by mrdooby** [comment]
For sure, thank you for your comments. I have implemented those changes.

**Pull request merged**